### PR TITLE
made update for rf oneAPI error handling and added bad args and kernels tests

### PR DESCRIPTION
**Pull request by makart19**
made update for rf oneAPI error handling and added bad args and kernels tests

### CODE REVIEW

File: cpp/oneapi/dal/algo/decision_forest/backend/cpu/infer_train_cls_kernel_test.cpp [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+    auto labels_table = result_infer.get_labels();
+    ASSERT_EQ(labels_table.has_data(), true);
+    ASSERT_EQ(labels_table.get_row_count(), row_count_test);
+    ASSERT_EQ(labels_table.get_column_count(), 1);
+    ASSERT_EQ(result_infer.get_probabilities().has_data(), true);
+    ASSERT_EQ(result_infer.get_probabilities().get_column_count(), class_count);
+    ASSERT_EQ(result_infer.get_probabilities().get_row_count(), row_count_test);
+
+    ASSERT_LE(calculate_classification_error(labels_table, y_test), accuracy_threshold);
```
Comment (michael-smirnov) [inline cpp/oneapi/dal/algo/decision_forest/backend/cpu/infer_train_cls_kernel_test.cpp:142]: Too many checks... Can we avoid to check something or separate this test?

File: cpp/oneapi/dal/algo/decision_forest/common.cpp [hunk 2]
```diff
@@ -16,6 +16,11 @@
 
 #include "oneapi/dal/algo/decision_forest/common.hpp"
 #include "oneapi/dal/algo/decision_forest/detail/model_impl.hpp"
+#include "oneapi/dal/exceptions.hpp"
+
+#define DAL_CHECK_DOMAIN_COND(cond, description) \
+    if (!(cond))                                 \
+        throw domain_error(description);
```
Comment (michael-smirnov) [inline cpp/oneapi/dal/algo/decision_forest/common.cpp:23]: Why you can't use function here?

File: cpp/oneapi/dal/algo/decision_forest/common_reg_test.cpp [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+*******************************************************************************/
+
+#include "gtest/gtest.h"
+#include "oneapi/dal/algo/decision_forest.hpp"
+#include "oneapi/dal/data/accessor.hpp"
+#include "oneapi/dal/data/table.hpp"
+
+//#include "oneapi/dal/algo/decision_forest/common_test.h"
+
+#include <cmath>
```
Comment (michael-smirnov) [inline cpp/oneapi/dal/algo/decision_forest/common_reg_test.cpp:24]: Where you need this?

File: cpp/oneapi/dal/algo/decision_forest/backend/cpu/infer_kernel_classification_dense.cpp [hunk 4]
```diff
@@ -58,8 +59,7 @@ static infer_result<Task> call_daal_kernel(const context_cpu& ctx,
 
     auto model_pimpl = dal::detail::pimpl_accessor().get_pimpl(trained_model);
     if (!model_pimpl->is_interop()) {
-        // throw exception
-        return infer_result<Task>();
+        throw dal::invalid_argument("Input model is inconsistent with kernel type");
```
[the remark this answers is not in the document]
Comment (rlnx) [reply cpp/oneapi/dal/algo/decision_forest/backend/cpu/infer_kernel_classification_dense.cpp:62]: I think, it's more internal error

File: cpp/oneapi/dal/algo/decision_forest/common.cpp (hunk 2 above)
Comment (rlnx) [reply cpp/oneapi/dal/algo/decision_forest/common.cpp:23]: We would need to have refactoring iteration and introduce internal API for validation like @makart19 used here

File: cpp/oneapi/dal/algo/decision_forest/backend/cpu/infer_kernel_classification_dense.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (makart19) [reply cpp/oneapi/dal/algo/decision_forest/backend/cpu/infer_kernel_classification_dense.cpp]: fixed

File: cpp/oneapi/dal/algo/decision_forest/backend/cpu/infer_kernel_regression_dense.cpp [hunk 5]
```diff
@@ -57,8 +57,7 @@ static infer_result<Task> call_daal_kernel(const context_cpu& ctx,
 
     auto model_pimpl = dal::detail::pimpl_accessor().get_pimpl(trained_model);
     if (!model_pimpl->is_interop()) {
-        // throw exception
-        return infer_result<Task>();
+        throw dal::invalid_argument("Input model is inconsistent with kernel type");
```
[the remark this answers is not in the document]
Comment (makart19) [reply cpp/oneapi/dal/algo/decision_forest/backend/cpu/infer_kernel_regression_dense.cpp]: fixed

File: cpp/oneapi/dal/algo/decision_forest/common_utils_test.hpp [hunk 6]
```diff
@@ -0,0 +1,94 @@
+/*******************************************************************************
```
[the remark this answers is not in the document]
Comment (makart19) [reply cpp/oneapi/dal/algo/decision_forest/common_utils_test.hpp]: placed into test folder

File: cpp/oneapi/dal/algo/decision_forest/backend/cpu/infer_train_cls_kernel_test.cpp [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+* limitations under the License.
+*******************************************************************************/
+
+#include "gtest/gtest.h"
+#include "oneapi/dal/algo/decision_forest.hpp"
+#include "oneapi/dal/algo/decision_forest/common_utils_test.hpp"
+#include "oneapi/dal/data/accessor.hpp"
+#include "oneapi/dal/data/table.hpp"
+
+#include <cmath>
```
[the remark this answers is not in the document]
Comment (makart19) [reply cpp/oneapi/dal/algo/decision_forest/backend/cpu/infer_train_cls_kernel_test.cpp]: thanks, removed

File: cpp/oneapi/dal/algo/decision_forest/common.cpp (hunk 2 above)
Comment (makart19) [reply cpp/oneapi/dal/algo/decision_forest/common.cpp]: replaced with function

File: cpp/oneapi/dal/algo/decision_forest/common_reg_test.cpp (hunk 3 above)
Comment (makart19) [reply cpp/oneapi/dal/algo/decision_forest/common_reg_test.cpp]: thanks, removed

**Pull request merged**